### PR TITLE
Alex/rollbackfix

### PR DESCRIPTION
**Pull request by alexanderlin**
Adds a test to check that if a transaction query fails early on in the transaction, then all other transaction SQL queries in the transaction will not affect the database. Also removed comments related to the transaction rollback fix which is also included allowing transaction to correctly rollback on first transaction query error. `assertRejects `was also imported in `deps.ts` for tehs test that checks if transaction throws an error on a malformed query.

### CODE REVIEW

File: Test/model_integration_tests/model_integration_test.ts [hunk 1]
```diff
@@ -161,5 +162,51 @@ describe('model methods', () => {
...
+        await Person.select().endTransaction();
+      } catch (_e) {
+        const yoda = await db.queryObject(
+          `SELECT name from people WHERE name = 'Yoda'`,
+        );
+        assertEquals(yoda.rows, [{ name: 'Yoda' }]);
+      }
+    });
+
+    it('it should not manipulate the database with any of the sql queries in transaction because of an invalid transactino query early on in the transaction', async () => {
```
Comment (Solit95) [inline Test/model_integration_tests/model_integration_test.ts:189]: transactino should be transaction

File: Test/model_integration_tests/model_integration_test.ts [hunk 2]
```diff
@@ -161,5 +162,51 @@ describe('model methods', () => {
...
+      try {
+        await Person.insert('name = Luke').transaction();
+        await Person.delete().where('name1 = Han Solo').transaction();
+        await Person.delete().where('name = Yoda').transaction;
```
Comment (Solit95) [inline Test/model_integration_tests/model_integration_test.ts:193]: transaction is not invoked, should it be?

File: Test/model_integration_tests/model_integration_test.ts [hunk 3]
```diff
@@ -161,5 +162,51 @@ describe('model methods', () => {
...
+        await Planet.delete().where('name = Luke Skywalker');
+        await Person.select().endTransaction();
```
Comment (Solit95) [inline Test/model_integration_tests/model_integration_test.ts:195]: seems a bit odd that we wouldnt just tack on endtransaction to planet.delete("luke skywalker"

File: Test/model_integration_tests/model_integration_test.ts [hunk 4]
```diff
@@ -161,5 +162,51 @@ describe('model methods', () => {
...
+        assertEquals(yoda.rows, [{ name: 'Yoda' }]);
+      }
+    });
+
+    it('it should not manipulate the database with any of the sql queries in transaction because of an invalid transactino query early on in the transaction', async () => {
+      try {
+        await Person.insert('name = Luke').transaction();
+        await Person.delete().where('name1 = Han Solo').transaction();
+        await Person.delete().where('name = Yoda').transaction;
+        await Planet.delete().where('name = Luke Skywalker');
```
Comment (rcheman) [inline Test/model_integration_tests/model_integration_test.ts:194]: Yoda and Luke Skywalker are not fully calling .transaction()

File: Test/model_integration_tests/model_integration_test.ts [hunk 5]
```diff
@@ -161,5 +162,51 @@ describe('model methods', () => {
       assertEquals(Shrike.rows, [{ name: 'Shrike' }]);
       assertEquals(Consul.rows, [{ name: 'The Consul' }]);
     });
+
+    it('it should throw an error because of a lack of a space around equal sign', async () => {
+      await assertRejects(async () => {
```
Comment (mjpg1) [inline Test/model_integration_tests/model_integration_test.ts:167]: a whole transaction should throw an error if a single query in the transaction is malformed

**Review by rcheman — APPROVED** [review]

**Review by Solit95 — APPROVED** [review]

**Pull request merged**